apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/AggregateQuerySnapshotTest.cs

```diff
@@ -23,54 +26,147 @@ public class AggregateQuerySnapshotTest
         var sampleReadTime = Timestamp.FromProto(CreateProtoTimestamp(1, 2));
-        int sampleCount = 10;
-        var control = new AggregateQuerySnapshot(aggQuery1, sampleReadTime, sampleCount);
+        var data = CreateData(20, 1.5, 10);
+        var equalData = CreateData(20, 1.5, 10);
+        var unequalData1 = CreateData(20, 1.5, 30);
+        var unequalData2 = CreateData(20, 1.5, 10);
+        unequalData2["Sum_other"] = CreateValue(20);
+        var control = new AggregateQuerySnapshot(aggQuery, sampleReadTime, data);
 
         EqualityTester.AssertEqual(control,
             equal: new[] {
                 // All the members are equal.
-                new AggregateQuerySnapshot(s_db.Collection("col").Count(), Timestamp.FromProto(CreateProtoTimestamp(1, 2)), 10)
+                new AggregateQuerySnapshot(aggQuery2, Timestamp.FromProto(CreateProtoTimestamp(1, 2)), equalData),
             },
             unequal: new[] {
                 // Unequal aggregate query
-                new AggregateQuerySnapshot(aggQuery2, sampleReadTime, sampleCount),
-                // Null count.
-                new AggregateQuerySnapshot(aggQuery2, sampleReadTime, null),
-                // Same aggregate query and count but different read time. 
-                new AggregateQuerySnapshot(s_db.Collection("col").Count(), Timestamp.FromProto(CreateProtoTimestamp(3, 4)), sampleCount),
-                // Same aggregate query but different read time and count.
-                new AggregateQuerySnapshot(s_db.Collection("col").Count(), Timestamp.FromProto(CreateProtoTimestamp(3, 4)), 20)
+                new AggregateQuerySnapshot(s_db.Collection("col").Count(), sampleReadTime, data),
+                // Same aggregate query but different read time
+                new AggregateQuerySnapshot(aggQuery, Timestamp.FromProto(CreateProtoTimestamp(3, 4)), data),
+                // Different data but for same keys
+                new AggregateQuerySnapshot(aggQuery, sampleReadTime, unequalData1),
+                // More keys (which probably wouldn't happen with the same aggregate query, but...)
+                new AggregateQuerySnapshot(aggQuery, sampleReadTime, unequalData2),
             });
     }
 
     [Fact]
-    public void DeserializationContextProperties()
+    public void GetValue()
+    {
+        var sampleReadTime = Timestamp.FromProto(CreateProtoTimestamp(1, 2));
+        var data = new MapField<string, Value>() { { "Sum_bar", CreateValue(20) } };
+        AggregateQuery aggQuery = s_db.Collection("col").Aggregate(Sum("bar"));
+        var aggQuerySnapshot = new AggregateQuerySnapshot(aggQuery, sampleReadTime, data);
+
+        // Alias key
+        Assert.Equal(20, aggQuerySnapshot.GetValue<long>("Sum_bar"));
+
+        // AggregateField key
+        Assert.Equal(20, aggQuerySnapshot.GetValue<long>(Sum("bar")));
+
+        // Object type output
+        Assert.Equal(20L, aggQuerySnapshot.GetValue<object>(Sum("bar")));
```

Comment:
This is probably sufficient to cover my previous comment about testing `snap.GetValue<object>(...)`, but I'll leave that comment in case you also want to add an integration test.

Reply:
My reason for adding an integration test would be as a reminder to ourselves that the "obvious" type isn't always used as the underlying result type.